src/sentry/api/endpoints/organization_discover_saved_queries.py

```diff
@@ -1,17 +1,65 @@
 from __future__ import absolute_import
 
-from sentry.api.bases.organization import OrganizationDiscoverPermission
-from sentry.api.bases import OrganizationEndpoint
-from sentry.models import DiscoverSavedQuery
+from rest_framework import serializers
+from rest_framework.response import Response
 
 from sentry.api.serializers import serialize
-from rest_framework.response import Response
+from sentry.api.serializers.rest_framework import ListField
+
+
+from sentry.api.bases.organization import OrganizationPermission
+from sentry.api.bases import OrganizationEndpoint
+from sentry.models import DiscoverSavedQuery
 
 from sentry import features
 
 
+class DiscoverSavedQueriesSerializer(serializers.Serializer):
+    name = serializers.CharField(required=True)
+    projects = ListField(
+        child=serializers.IntegerField(),
+        required=True,
+        allow_null=False,
+    )
+    start = serializers.DateTimeField(required=False)
+    end = serializers.DateTimeField(required=False)
+    range = serializers.CharField(required=False)
+    fields = ListField(
+        child=serializers.CharField(),
+        required=False,
+        allow_null=True,
+    )
+    limit = serializers.IntegerField(min_value=0, max_value=1000, required=False)
+    rollup = serializers.IntegerField(required=False)
+    orderby = serializers.CharField(required=False)
+    conditions = ListField(
+        child=ListField(),
+        required=False,
+        allow_null=True,
+    )
+    aggregations = ListField(
+        child=ListField(),
+        required=False,
+        allow_null=True,
+        default=[]
+    )
+    groupby = ListField(
+        child=serializers.CharField(),
+        required=False,
+        allow_null=True,
+    )
+
+    def validate(self, data):
+        return {
+            'name': data['name'],
+            'project_ids': data['projects'],
+            'query': {
+            }
+        }
```

Comment:
Is there more coming here?

Reply:
Oops, yes, pushed the rest of the changes